### PR TITLE
Do not close channel after subscription cancellation

### PR DESCRIPTION
**Pull request by jbowen93**

### CODE REVIEW

**Comment by tzdybal** [comment]
Rebased PR on top of current main. Closing a channel should be used for signaling, so in original code it was used incorrectly. (`Close` is not required to free resources, it's done by GC anyways).

**Review by tzdybal — APPROVED** [review]

File: rpc/json/ws.go [hunk 1]
```diff
@@ -68,7 +68,8 @@ func (h *handler) wsHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 		if mt != websocket.TextMessage {
-			h.logger.Error("expected text message")
+			// TODO(tzdybal): https://github.com/celestiaorg/optimint/issues/465`
```
Comment (adlerjohn) [inline rpc/json/ws.go:71]:
```suggestion
			// TODO(tzdybal): https://github.com/celestiaorg/optimint/issues/465
```

Comment (adlerjohn) [inline rpc/json/ws.go:71]: This looks orthogonal to this PR, can you do it in a separate PR?

Comment (tzdybal) [reply rpc/json/ws.go]: https://github.com/celestiaorg/optimint/pull/469
